packages/google_sign_in/google_sign_in_ios/ios/Classes/FLTGoogleSignInPlugin.m

```diff
@@ -78,52 +79,45 @@ - (instancetype)initWithSignIn:(GIDSignIn *)signIn {
-    if (path) {
-      NSMutableDictionary<NSString *, NSString *> *plist =
-          [[NSMutableDictionary alloc] initWithContentsOfFile:path];
-      BOOL hasDynamicClientId = [call.arguments[@"clientId"] isKindOfClass:[NSString class]];
-
-      if (hasDynamicClientId) {
-        self.signIn.clientID = call.arguments[@"clientId"];
-      } else {
-        self.signIn.clientID = plist[kClientIdKey];
-      }
-
-      self.signIn.serverClientID = plist[kServerClientIdKey];
-      self.signIn.scopes = call.arguments[@"scopes"];
-      if (call.arguments[@"hostedDomain"] == [NSNull null]) {
-        self.signIn.hostedDomain = nil;
-      } else {
-        self.signIn.hostedDomain = call.arguments[@"hostedDomain"];
+    GIDConfiguration *configuration =
+        [self configurationClientIdArgument:call.arguments[@"clientId"]
+                       hostedDomainArgument:call.arguments[@"hostedDomain"]];
+    if (configuration != nil) {
+      if ([call.arguments[@"scopes"] isKindOfClass:[NSArray class]]) {
+        self.requestedScopes = [NSSet setWithArray:call.arguments[@"scopes"]];
       }
+      self.configuration = configuration;
       result(nil);
     } else {
       result([FlutterError errorWithCode:@"missing-config"
                                  message:@"GoogleService-Info.plist file not found"
                                  details:nil]);
     }
   } else if ([call.method isEqualToString:@"signInSilently"]) {
-    if ([self setAccountRequest:result]) {
-      [self.signIn restorePreviousSignIn];
-    }
+    [self.signIn restorePreviousSignInWithCallback:^(GIDGoogleUser *user, NSError *error) {
+      [self didSignInForUser:user result:result withError:error];
+    }];
   } else if ([call.method isEqualToString:@"isSignedIn"]) {
     result(@([self.signIn hasPreviousSignIn]));
   } else if ([call.method isEqualToString:@"signIn"]) {
-    self.signIn.presentingViewController = [self topViewController];
-
-    if ([self setAccountRequest:result]) {
-      @try {
-        [self.signIn signIn];
-      } @catch (NSException *e) {
-        result([FlutterError errorWithCode:@"google_sign_in" message:e.reason details:e.name]);
-        [e raise];
-      }
+    @try {
+      GIDConfiguration *configuration =
+          self.configuration ?: [self configurationClientIdArgument:nil hostedDomainArgument:nil];
+      [self.signIn signInWithConfiguration:configuration
+                  presentingViewController:[self topViewController]
+                                      hint:nil
+                          additionalScopes:self.requestedScopes.allObjects
+                                  callback:^(GIDGoogleUser *user, NSError *error) {
+                                    [self didSignInForUser:user result:result withError:error];
+                                  }];
+    } @catch (NSException *e) {
+      result([FlutterError errorWithCode:@"google_sign_in" message:e.reason details:e.name]);
+      [e raise];
```

Comment:
Even though this looks like a large change, I tried to change as little as possible, and this was moved from
https://github.com/flutter/plugins/blob/f25fc10f93018ee15a4906f91ff314ae129c3223/packages/google_sign_in/google_sign_in_ios/ios/Classes/FLTGoogleSignInPlugin.m#L116-L121

This was introduced in #1180:
> We would still like to crash the app since the crash would be a good reminder for the users to set up any necessary properties to use this plugin, and the users should not ship the product without fixing the issue that causes the exception.